src/reflect/value.go

```diff
@@ -882,8 +882,43 @@ func (v Value) Convert(t Type) Value {
 	panic("unimplemented: (reflect.Value).Convert()")
 }
 
+//go:linkname slicePanic runtime.slicePanic
+func slicePanic()
+
 func MakeSlice(typ Type, len, cap int) Value {
-	panic("unimplemented: reflect.MakeSlice()")
+	if typ.Kind() != Slice {
+		panic("reflect.MakeSlice of non-slice type")
+	}
+
+	ulen := uintptr(len)
+	ucap := uintptr(cap)
+	if ulen > ucap {
+		slicePanic()
+	}
+
+	size, ok := mulNoOverflow(typ.Size(), ucap)
```

Comment:
You need to use the element size here, not the slice size (which is always `3 * unsafe.Size(uintptr(0))`).

```suggestion
	size, ok := mulNoOverflow(typ.Elem().Size(), ucap)
```